File: app/src/main/java/com/tobikster/medicreminder/di/ViewModelSubComponent.java
package com.tobikster.medicreminder.di;

import com.tobikster.medicreminder.ui.reminders.model.ReminderDetailsModel;
import com.tobikster.medicreminder.ui.reminders.model.RemindersListModel;

import dagger.Subcomponent;

@Subcomponent
public interface ViewModelSubComponent {
	@Subcomponent.Builder
	interface Builder {
		ViewModelSubComponent build();
	}

	RemindersListModel remindersListModel();
	ReminderDetailsModel reminderDetailsModel();
}

File: app/src/main/java/com/tobikster/medicreminder/di/ViewModelSubComponent.kt
package com.tobikster.medicreminder.di

import com.tobikster.medicreminder.ui.reminders.model.ReminderDetailsModel
import com.tobikster.medicreminder.ui.reminders.model.RemindersListModel

import dagger.Subcomponent

@Subcomponent
interface ViewModelSubComponent {
	@Subcomponent.Builder
	interface Builder {
		fun build(): ViewModelSubComponent
	}

	fun remindersListModel(): RemindersListModel
	fun reminderDetailsModel(): ReminderDetailsModel
}
